File: fibonacci.py
from __future__ import print_function

def fibonacci(N):
	n, m = 1, 0
	for i in range(N):
		n = n + m
		m = n - m
	return n

def shell():
	"""
	
	Console Function to create the interactive Shell.
	
	"""
	print("Welcome to Fibonacci Sequence Calculator")
	
	while True:
		print(">>> ", end='')
		entry = input()
		if entry == "quit":
			break
		if not entry.isdigit():
			print("Please enter a postive integer.")
		else:
			print(fibonacci(int(entry)))


if __name__ == "__main__":
	shell()

File: fibonacci.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

unsigned long int fibonacci(int N) {
	int n, m, i = 1, 0, 0;
	while (i < N) {
		n = n + m;
		m = n - m;
		i++;
	}
	return n;
}

void main(void) {
	printf("Welcome to the interactive Fibonacci Sequence calculator\n");
	char N[10], quit[10];
	
	while (1) {
	 	printf(">>> ");
	 	scanf(" %s", &N);
	 	strcpy(quit, "quit")
	 	
	 	if (!(strcmp(quit, "quit"))) {
	 		break;
	 	} else if (!(isdigit(N))) {
	 		printf("Please enter a positive integer.\n");
	 	} else {
	 		printf("%ld\n", fibonacci(N));		
	 	}
	}
}
